web/src/components/users/FirstUserForm.tsx

```diff
@@ -80,84 +78,78 @@ const UsernameSuggestions = ({
   );
 };
 
-type FormState = {
-  load?: boolean;
-  user?: FirstUser;
-  isEditing?: boolean;
-};
-
 // TODO: create an object for errors using the input name as key and show them
 // close to the related input.
 // TODO: extract the suggestions logic.
 export default function FirstUserForm() {
   const firstUser = useFirstUser();
   const setFirstUser = useFirstUserMutation();
-  const [state, setState] = useState<FormState>({});
+  const [fullName, setFullName] = useState(firstUser?.fullName);
+  const [userName, setUserName] = useState(firstUser?.userName);
+  const [password, setPassword] = useState(
+    firstUser && firstUser.hashedPassword ? "" : firstUser.password,
+  );
   const [errors, setErrors] = useState([]);
+  const [usingHashedPassword, setUsingHashedPassword] = useState(
+    firstUser ? firstUser.hashedPassword : false,
+  );
   const [showSuggestions, setShowSuggestions] = useState(false);
   const [insideDropDown, setInsideDropDown] = useState(false);
   const [focusedIndex, setFocusedIndex] = useState(-1);
   const [suggestions, setSuggestions] = useState([]);
-  const [changePassword, setChangePassword] = useState(true);
+  const [changePassword, setChangePassword] = useState(firstUser?.userName === "");
   const usernameInputRef = useRef<HTMLInputElement>();
   const navigate = useNavigate();
   const passwordRef = useRef<HTMLInputElement>();
 
-  useEffect(() => {
-    const editing = firstUser.userName !== "";
-    setState({
-      load: true,
-      user: firstUser,
-      isEditing: editing,
-    });
-    setChangePassword(!editing);
-  }, [firstUser]);
-
   useEffect(() => {
     if (showSuggestions) {
       setFocusedIndex(-1);
     }
   }, [showSuggestions]);
 
-  if (!state.load) return <Loading />;
+  if (!firstUser) return <Loading />;
+
+  const isEditing = firstUser.userName !== "";
 
   const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     setErrors([]);
-
+    const nextErrors = [];
     const passwordInput = passwordRef.current;
-    const formData = new FormData(e.currentTarget);
-    const user: Partial<FirstUser> & { passwordConfirmation?: string } = {};
-    // FIXME: have a look to https://www.patternfly.org/components/forms/form#form-state
-    formData.forEach((value, key) => {
-      user[key] = value;
-    });
 
-    if (!changePassword) {
-      delete user.password;
-    } else {
-      // the web UI only supports plain text passwords, this resets the flag if a hashed
-      // password was previously set from CLI
-      user.hashedPassword = false;
+    const data: Partial<FirstUser> = {
+      fullName,
+      userName,
+      // FIXME: Stop sending the password conditionally (requires backend changes)
```

Comment:
Note to myself: this means to send always the firstUser password once the backend allows to use it. @imobachgs  is working on it.